usr/src/cmd/text-install/osol_install/text_install/zpool_selection.py

```diff
@@ -0,0 +1,319 @@
+        self.be_name_edit = EditField(self.edit_area,
+                                      window=self.be_name_list,
+                                      validate=be_name_valid,
+                                      error_win=self.be_name_err,
+                                      text=self.install_profile.be_name)
+         
+        y_loc += 2
+        boot_configuration_width = textwidth(ZpoolScreen.OVERWRITE_BOOT_CONFIGURATION_LABEL) + 5
+        cols = (self.win_size_x - boot_configuration_width) / 2
+        boot_configuration_area = WindowArea(1, boot_configuration_width, y_loc, cols)
+
+        self.boot_configuration_item = MultiListItem(boot_configuration_area,
+                                                     window=self.center_win,
+                                                     text=ZpoolScreen.OVERWRITE_BOOT_CONFIGURATION_LABEL,
+                                                     used=self.install_profile.overwrite_boot_configuration)
+        
+        self.boot_configuration_item.on_select = on_select_obc
+        self.boot_configuration_item.on_select_kwargs["pool_select"] = self
+
+        self.main_win.do_update()
+        self.center_win.activate_object(self.pool_win)
+        self.pool_win.activate_object(self.selected_pool)
+        # Set the flag so that the pool is not copied by on_change_screen,
+        # unless on_activate gets called as a result of the user changing
+        # the selected pool.
+        self.do_copy = False
+
+    def on_change_screen(self):
+        ''' Assign the selected pool to the InstallProfile, and make note of
+        its index (in case the user returns to this screen later)
+        
+        '''
+        if self.do_copy or self.install_profile.pool_name is None:
+            self.install_profile.pool_name = self.existing_pools[self.pool_win.active_object]
+        self.selected_pool = self.pool_win.active_object
+        self.install_profile.be_name = self.be_name_edit.get_text()
+
+    def validate(self):
+        pool_name = self.existing_pools[self.pool_win.active_object]
+        be_name = self.be_name_edit.get_text()
+        if not be_name:
+            raise UIMessage, ZpoolScreen.BE_NAME_EMPTY_ERROR
+
+        be_names = get_zpool_be_names(pool_name)
+        if be_name in be_names:
+            filesystem_dict = {"pool_name": pool_name, "be_name": be_name}
+            raise UIMessage, ZpoolScreen.FILESYSTEM_EXISTS_ERROR % filesystem_dict
+
+def on_activate(pool_select=None):
+    '''When a disk is selected, pass its data to the disk_select_screen'''
+
+    # User selected a different disk; set the flag so that it gets copied
+    # later
+    pool_select.do_copy = True
+
+def be_name_valid(edit_field):
+    '''Ensure BE name is valid'''
+
+    be_name = edit_field.get_text()
+    if not be_name:
+        raise UIMessage, ZpoolScreen.BE_NAME_EMPTY_ERROR
+    
+    search = re.compile(r'[^a-zA-Z0-9_\-]').search
```

Comment:
As I understand from the docs (https://docs.python.org/2/library/re.html), it should be.
(\w is equivalent to [^a-zA-Z0-9_] when re.LOCALE is not set).